accessibilityTest/AccessibilityQuickview.ts

```diff
@@ -1,33 +1,101 @@
+import { $$, Quickview, Component, get, Dom } from 'coveo-search-ui';
+import { afterQuerySuccess, getRoot, testResultElement } from './Testing';
+
+function nodeListToArray<T extends Node>(nodeList: NodeListOf<T>) {
+  const result: T[] = [];
+  for (let i = 0; i < nodeList.length; i++) {
+    result.push(nodeList.item(i));
+  }
+  return result;
+}
+
+function getFocusableElements() {
+  return nodeListToArray<HTMLElement>(
+    document.querySelectorAll(`
+      a[href]:not([tabindex='-1']),
+      area[href]:not([tabindex='-1']),
+      input:not([disabled]):not([tabindex='-1']),
+      select:not([disabled]):not([tabindex='-1']),
+      textarea:not([disabled]):not([tabindex='-1']),
+      button:not([disabled]):not([tabindex='-1']),
+      iframe:not([tabindex='-1']),
+      [tabindex]:not([tabindex='-1']),
+      [contentEditable=true]:not([tabindex='-1'])
+    `)
+  ).sort((a, b) => a.tabIndex - b.tabIndex);
+}
+
+function getNextFocusableElement() {
+  const elements = getFocusableElements();
+  const currentIndex = document.activeElement ? elements.indexOf(document.activeElement as HTMLElement) : -1;
+  return elements[(currentIndex + 1) % elements.length];
+}
+
+function simulateTabKey() {
+  const event = new KeyboardEvent('keydown');
+  Object.defineProperty(event, 'keyCode', { get: () => 9 });
+  document.body.dispatchEvent(event);
+  if (!event.defaultPrevented) {
+    getNextFocusableElement().focus();
+  }
+}
 
 export const AccessibilityQuickview = () => {
   describe('Quickview', () => {
-    it('should be accessible', async done => {
-      const quickviewElement = $$('div', {
+    async function openQuickview() {
+      await (get(document.querySelector(`.${Component.computeCssClassName(Quickview)}`)) as Quickview).open();
+    }
+
+    let quickviewElement: Dom;
+    beforeEach(async done => {
+      quickviewElement = $$('div', {
         className: Component.computeCssClassName(Quickview)
       });
 
       testResultElement(quickviewElement.el);
       await afterQuerySuccess();
+      done();
+    });
+
+    it('should be accessible', async done => {
       const axeResults = await axe.run(getRoot());
       expect(axeResults).toBeAccessible();
       done();
     });
 
     it('should be accessible when opened', async done => {
-      const quickviewElement = $$('div', {
-        className: Component.computeCssClassName(Quickview)
-      });
-
-      testResultElement(quickviewElement.el);
-      await afterQuerySuccess();
-      (get(document.querySelector(`.${Component.computeCssClassName(Quickview)}`)) as Quickview).open();
-      await afterDelay(500);
+      await openQuickview();
       const axeResults = await axe.run(getRoot());
       expect(axeResults).toBeAccessible();
       done();
     });
+
+    it('should have a close button when opened', async done => {
+      await openQuickview();
+      expect(document.querySelector('.coveo-small-close')).not.toBeNull();
```

Comment:
I would create a function for `document.querySelector('.coveo-small-close')` -> `getQuickviewCloseButton()`

Reply:
Good idea, fixing that.